src/Components/Facility/ConsultationClaims.tsx

```diff
@@ -19,48 +20,46 @@ export default function ConsultationClaims({
   facilityId,
   consultationId,
   patientId,
-}: Props) {
-  const dispatch = useDispatch<any>();
-  const [claims, setClaims] = useState<HCXClaimModel[]>();
+}: IConsultationClaimsProps) {
+  const { t } = useTranslation();
+
   const [isCreateLoading, setIsCreateLoading] = useState(false);
 
-  const fetchClaims = useCallback(async () => {
-    const res = await dispatch(
-      HCXActions.claims.list({
+  const { data: claimsResult, refetch: refetchClaims } = useQuery(
+    routes.hcx.claims.list,
+    {
+      query: {
         ordering: "-modified_date",
         consultation: consultationId,
-      }),
-    );
-
-    if (res.data && res.data.results) {
-      setClaims(res.data.results);
-      if (isCreateLoading)
-        Notification.Success({ msg: "Fetched Claim Approval Results" });
-    } else {
-      if (isCreateLoading)
-        Notification.Success({ msg: "Error Fetched Claim Approval Results" });
-    }
-    setIsCreateLoading(false);
-  }, [dispatch, consultationId]);
-
-  useEffect(() => {
-    fetchClaims();
-  }, [fetchClaims]);
+      },
+      onResponse: (res) => {
+        if (res.data && res.data.results) {
+          if (isCreateLoading)
+            Notification.Success({ msg: "Fetched Claim Approval Results" });
+        } else {
+          if (isCreateLoading)
+            Notification.Error({
+              msg: "Error Fetching Claim Approval Results",
+            });
+        }
+      },
```

Comment:
i18n + reducing if else ladder for readability.

```suggestion
        if (!isCreateLoading) {
          return;
        }
        
        if (res.data && res.data.results) {
          Notification.Success({ msg: "Fetched Claim Approval Results" });
          return;
        }
        
        Notification.Error({ msg: "Error Fetching Claim Approval Results" });
      },
```